pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/api/go/profile"
+	"github.com/lf-edge/eve/pkg/pillar/flextimer"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+const (
+	defaultLocalProfileServerPort = "8888"
+	savedLocalProfileFile         = "lastlocalprofile"
+)
+
+// urlAndSrcIP structure for mapping source IP and url of local server
+type urlAndSrcIP struct {
+	srcIP     net.IP
+	actualURL string
+}
+
+func getLocalProfileURL(localProfileServer string) (string, error) {
+	localProfileURL := fmt.Sprintf("http://%s", localProfileServer)
+	u, err := url.Parse(localProfileURL)
+	if err != nil {
+		return "", fmt.Errorf("url.Parse: %s", err)
+	}
+	if u.Port() == "" {
+		localProfileURL = fmt.Sprintf("%s:%s", localProfileURL, defaultLocalProfileServerPort)
+	}
+	return fmt.Sprintf("%s/api/v1/local_profile", localProfileURL), nil
+}
+
+//checkAndHandleLocalProfileUpdate gets local profile config if localProfileServer defined,
+//determines currentProfile and notify on changes
+func checkAndHandleLocalProfileUpdate(getconfigCtx *getconfigContext) {
+	localProfileServer := getconfigCtx.localProfileServer
+	if localProfileServer == "" {
+		return
+	}
+	localProfileURL, err := getLocalProfileURL(localProfileServer)
+	if err != nil {
+		log.Errorf("checkAndHandleLocalProfileUpdate getLocalProfileURL: %s", err)
+		return
+	}
+	localProfileConfig, err := getLocalProfileConfig(localProfileURL, getconfigCtx)
+	if err != nil {
+		log.Errorf("checkAndHandleLocalProfileUpdate getLocalProfileConfig: %s", err)
+		return
+	}
+	localProfile := localProfileConfig.GetLocalProfile()
+	if getconfigCtx.localProfile != localProfile {
+		log.Noticef("checkAndHandleLocalProfileUpdate: local profile changed from %s to %s",
+			getconfigCtx.localProfile, localProfile)
+		getconfigCtx.localProfile = localProfile
+	}
+	checkAndHandleCurrentProfileUpdate(getconfigCtx)
+	writeReceivedProtoMessageLocalProfile(localProfileConfig)
```

Comment:
Why do you need to call these when the localProfile didn't change?

Reply:
Good point. Will move it into if and put in else branch touch of file date